tagtagtag-sound branch: tagtagtag-mixerd compilation fails on Debian Bullseye (with gcc 10)

(test done with _raspios_lite:2021-10-30_ base image)
```
Cloning into 'wm8960'...
cc -Wall -Werror -lasound -lpthread tagtagtag-mixerd.c -o tagtagtag-mixerd
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `hctl_jack_cb':
tagtagtag-mixerd.c:(.text+0x94): undefined reference to `snd_hctl_elem_get_callback_private'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x9c): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xcc): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xf0): undefined reference to `snd_hctl_elem_read'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x100): undefined reference to `snd_ctl_elem_value_get_boolean'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `hctl_button_cb':
tagtagtag-mixerd.c:(.text+0x138): undefined reference to `snd_hctl_elem_get_callback_private'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x140): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x170): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x194): undefined reference to `snd_hctl_elem_read'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x1a4): undefined reference to `snd_ctl_elem_value_get_integer'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x1c8): undefined reference to `snd_ctl_elem_value_get_integer'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `setup_hctl':
tagtagtag-mixerd.c:(.text+0x9dc): undefined reference to `snd_hctl_open'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xa20): undefined reference to `snd_hctl_nonblock'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xa60): undefined reference to `snd_hctl_load'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `teardown_hctl':
tagtagtag-mixerd.c:(.text+0xacc): undefined reference to `snd_hctl_close'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `setup_callbacks':
tagtagtag-mixerd.c:(.text+0xaf4): undefined reference to `snd_ctl_elem_id_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb24): undefined reference to `snd_ctl_elem_id_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb48): undefined reference to `snd_ctl_elem_id_set_interface'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb58): undefined reference to `snd_ctl_elem_id_set_name'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb68): undefined reference to `snd_hctl_find_elem'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb78): undefined reference to `snd_hctl_elem_set_callback'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xb84): undefined reference to `snd_hctl_elem_set_callback_private'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xba0): undefined reference to `snd_ctl_elem_id_set_name'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xbb0): undefined reference to `snd_hctl_find_elem'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xbc0): undefined reference to `snd_hctl_elem_set_callback'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xbcc): undefined reference to `snd_hctl_elem_set_callback_private'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `set_ctl_elem_value':
tagtagtag-mixerd.c:(.text+0xc14): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xc44): undefined reference to `snd_ctl_elem_value_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xc5c): undefined reference to `snd_ctl_elem_id_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xc8c): undefined reference to `snd_ctl_elem_id_sizeof'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xcb0): undefined reference to `snd_ctl_elem_id_set_interface'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xcc0): undefined reference to `snd_ctl_elem_id_set_name'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xcd0): undefined reference to `snd_hctl_find_elem'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd08): undefined reference to `snd_ctl_elem_value_set_boolean'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd20): undefined reference to `snd_ctl_elem_value_set_boolean'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd34): undefined reference to `snd_ctl_elem_value_set_boolean'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd4c): undefined reference to `snd_ctl_elem_value_set_integer'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd64): undefined reference to `snd_ctl_elem_value_set_integer'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd78): undefined reference to `snd_ctl_elem_value_set_integer'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0xd8c): undefined reference to `snd_hctl_elem_write'
/usr/bin/ld: /tmp/ccNWcwfx.o: in function `main':
tagtagtag-mixerd.c:(.text+0x1274): undefined reference to `pthread_sigmask'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x14c4): undefined reference to `snd_hctl_poll_descriptors_count'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x1514): undefined reference to `snd_hctl_poll_descriptors'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x15f8): undefined reference to `snd_hctl_poll_descriptors_revents'
/usr/bin/ld: tagtagtag-mixerd.c:(.text+0x1644): undefined reference to `snd_hctl_handle_events'
collect2: error: ld returned 1 exit status
make: *** [Makefile:44: tagtagtag-mixerd] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -41,10 +41,10 @@ install: snd-soc-wm8960.ko snd-soc-max9759.ko snd-soc-volume-gpio.ko tagtagtag-s
 	systemctl enable tagtagtag-mixerd
 
 tagtagtag-mixerd : tagtagtag-mixerd.c
-	cc -Wall -Werror -lasound -lpthread $< -o $@
+	cc -Wall -Werror $< -lasound -lpthread -o $@
 
 tagtagtag-mixerd-test : tagtagtag-mixerd.c
-	cc -Wall -Werror -lasound -lpthread -DTEST $< -o $@
+	cc -Wall -Werror -DTEST $< -lasound -lpthread -o $@
 
 test : tagtagtag-mixerd-test
 	./tagtagtag-mixerd-test

